taca/backup/backup.py

```diff
@@ -0,0 +1,309 @@
+        Given 'out_file' is always used for the later cmd and also stdout can be return
+        for the later cmd. In case of failure, the 'tmp_files' are removed"""
+        if out_file:
+            if not cmd2:
+                stdout1 = open(out_file, 'w')
+            else:
+                stdout1 = sp.PIPE
+                stdout2 = open(out_file, 'w')
+        else:
+            stdout1 = sp.PIPE
+            stdout2 = sp.PIPE
+        # calling the commands
+        try:
+            cmd1 = cmd1.split()
+            p1 = sp.Popen(cmd1, stdout=stdout1, stderr=sp.PIPE)
+            if cmd2:
+                cmd2 = cmd2.split()
+                p2 = sp.Popen(cmd2, stdin=p1.stdout, stdout=stdout2, stderr=sp.PIPE)
+                p2_stat = p2.wait()
+                p2_out, p2_err = p2.communicate()
+                if not self._check_status(cmd2, p2_stat, p2_err, tmp_files):
+                    return False
+            p1_stat = p1.wait()
+            p1_out, p1_err = p1.communicate()
+            if not self._check_status(cmd1, p1_stat, p1_err, tmp_files):
+                return False
+        except Exception, e:
+            raise e
+        finally:
+            if out_file:
+                if not cmd2:
+                    stdout1.close()
+                else:
+                    stdout2.close()
+            if return_out:
+                if cmd2:
+                    return (True, p2_out)
+                return (True, p1_out)
+            return True
+
+    def _check_status(self, cmd, status, err_msg, files_to_remove=[]):
+        """Check if a subprocess status is success and log error if failed"""
+        if status != 0:
+            self._clean_tmp_files(files_to_remove)
+            logger.error("Command '{}' failed with the error '{}'".format(" ".join(cmd),err_msg))
+            return False
+        return True
+
+    def _clean_tmp_files(self, files):
+        """Remove the file is exist"""
+        for fl in files:
+            if os.path.exists(fl):
+                os.remove(fl)
+
+    @classmethod
+    def encrypt_runs(cls, run, force):
+        """Encrypt the runs that have been collected"""
+        bk = cls(run)
+        bk.collect_runs(ext=".tar.gz")
+        logger.info("In total, found {} run(s) to be encrypted".format(len(bk.runs)))
+        for run in bk.runs:
+            run.flag = "{}.encrypting".format(run.name)
+            run.dst_key_encrypted = os.path.join(bk.keys_path, run.key_encrypted)
+            tmp_files = [run.zip_encrypted, run.key_encrypted, run.key, run.flag]
+            logger.info("Encryption of run {} is now started".format(run.name))
+            # Check if there is enough space
+            if not bk.avail_disk_space(path=run.path, run=run.name):
+                logger.error("There is no enough disk space for compression, kindly check and archive encrypted runs")
+                raise SystemExit
```

Comment:
Can you send a mail in this case. You can check how the mail is sent in taca demux. I think it is important to get informed about this